File: Programs/insertion-sort.py
def insSort(arr):
    for i in range(1,len(arr)):  #we initiate it from 1 as we assume 0th element as sorted.
        key=arr[i] #assign the key
        j=i-1
        while key<arr[j] and j>=0: #this loop shifts all the elements greater than key
            arr[j+1]=arr[j]
            j-=1
        arr[j+1]=key #after shifting prosess this give the key its right position.
    return arr
    
    
x=[2,7,3,9,1,6]
print(insSort(x))

File: Programs/C,C++/InsertionSort.c
#include<stdio.h>
void insSort(int arr[],int n)
{
    int i,j,key;
    for(j=1;j<n;j++)
    {
        key=arr[j];
        i=j-1;
        while (i>=0 && arr[i]>key)
        {
            arr[i+1]=arr[i];
            i=i-1;
        }
        arr[i+1]=key;
    }
}
int main()
{
    int arr[10],n,i;
    printf("\n Enter the number of elements: ");
    scanf("%d",&n);
    printf("\n Enter the array: \n");
    for(i=0;i<n;i++)
    {
        scanf("%d",&arr[i]);
    }   
    printf("\n The inserted array is: ");
    for(i=0;i<n;i++)
        printf("%d ",arr[i]);
    insSort(arr,n);
    printf("\n The sorted array is: ");
    for(i=0;i<n;i++)
        printf("%d ",arr[i]);
    return 0;    
}
